contracts/implementation/Resolver.sol

```diff
@@ -0,0 +1,24 @@
+//SPDX-License-Identifier: CC-BY-NC-ND-4.0
+pragma solidity 0.8.7;
+
+interface IPoolKeeper {
+    function checkUpkeepSinglePool(address pool) external view returns (bool);
+}
+
+contract Resolver {
+    address public immutable PoolKeeper = 0x759E817F0C40B11C775d1071d466B5ff5c6ce28e;
```

Comment:
I reckon this should be set in the constructor, since this could change per deployment.

It should also be `poolKeeper` (camelCase)